src/darwin/Framework/CHIP/MTRDevice.h

```diff
@@ -201,6 +201,26 @@ typedef NS_ENUM(NSUInteger, MTRDeviceState) {
                                       completion:(MTRDeviceOpenCommissioningWindowHandler)completion
     MTR_AVAILABLE(ios(17.0), macos(14.0), watchos(10.0), tvos(17.0));
 
+/**
+ * Requests the device to transfer the diagnostic logs and return an array of file path(s) in the completion handler
+ * based on the requested log types in the logTypeBitmask.
+ *
+ * Logs can be of types - end user support, network diagnostic and crash logs.
+ *
+ * @param logTypeBitmask    The log type encoded in a bitmask.
+ * @param timeout           The timeout in seconds by which the logs must be retreived. If we exceed the timeout, the completion handler
+ *                          will be called with the file path array set to null.
+ * @param clientQueue       The client queue that the caller passes in on which the completion handler will be called.
+ * @param completionHandler The completion handler that will be called to pass in the file path(s) for the type(s) of logs requested
+ *
+ *
+ */
+- (void)getDiagnosticLogs:(MTRDevice *)device
```

Comment:
The MTRDevice is `self` here, no?  I think this should be more like:
```
- (void)getDiagnosticLogsOfTypes:(NeedAnNSOptionsTypeHere)types
```

and define an NS_OPTIONS for the types.  We can then document what the values mean in there.

Reply:
done.